src/libraries/System.Net.Requests/src/System/Net/RequestStreamContent.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net
+{
+    internal sealed class RequestStreamContent(HttpClientContentStream requestStream) : HttpContent
+    {
+        protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
+        {
+            return SerializeToStreamAsync(stream, context, default);
+        }
+        protected override async Task SerializeToStreamAsync(Stream stream, TransportContext? context, CancellationToken cancellationToken)
+        {
+            Debug.Assert(requestStream is not null);
+            Debug.Assert(stream is not null);
+
+            await stream.FlushAsync(cancellationToken).ConfigureAwait(false);
+            await requestStream.CopyToAsync(stream, requestStream.GetBuffer().ReadBytesAvailable, cancellationToken).ConfigureAwait(false);
```

Comment:
This is copying data from `requestStream` to `stream` and one chunk of size `ReadBytesAvailable`. Is this what you really want?

Reply:
Looking at framework code, looks like each write to `ConnectStream` (in our case `RequestStream`) is one chunk. This is not behaving like that. Or am I missing something?

Reply:
I think we should drop the buffer size argument here and let Stream use its default.

Reply:
If we were to keep the chunk size corresponding to how big byte[] user wrote to the request stream. I think we need to implement the copying ourselves and not rely on `CopyToAsync`. Each write to the underlying stream will create one chunk, so we should propagate writes 1:1.